disnake/ext/commands/common_bot_base.py

```diff
@@ -64,14 +51,16 @@ def __init__(
     ) -> None:
         self.__cogs: Dict[str, Cog] = {}
         self.__extensions: Dict[str, types.ModuleType] = {}
-        self.extra_events: Dict[str, List[CoroFunc]] = {}
         self._is_closed: bool = False
 
         self.owner_id: Optional[int] = owner_id
         self.owner_ids: Set[int] = owner_ids or set()
         self.owner: Optional[disnake.User] = None
         self.owners: Set[disnake.TeamMember] = set()
 
+        if TYPE_CHECKING:
+            self.extra_events: Dict[str, List[CoroFunc]]
```

Comment:
another nit, it's common practice to place typing-only declarations at the class level and not inside `__init__`